src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1906,26 +1982,20 @@ class Simplex_tree {
     bool modified = false;
```

Comment:
I performed [this benchmark](https://gist.github.com/VincentRouvreau/a2df412c4369098d7cfc9f6274f1c843#file-gistfile1-txt) and it seems this implementation is even faster than `master`'s one:
# master
```txt
(gudhi-devel) gailuron@precision:~/workspace/gudhi/gudhi-devel/build/src/Simplex_tree/benchmark$ make make_filtration_non_decreasing && ./make_filtration_non_decreasing
Building CXX object src/Simplex_tree/benchmark/CMakeFiles/make_filtration_non_decreasing.dir/make_filtration_non_decreasing.cpp.o
Linking CXX executable make_filtration_non_decreasing
Built target make_filtration_non_decreasing
num_simplices= 3799085
... make_filtration_non_decreasing: 0.712s
... make_filtration_non_decreasing: 0.711s
... make_filtration_non_decreasing: 0.71s
... make_filtration_non_decreasing: 0.712s
... make_filtration_non_decreasing: 0.712s
(gudhi-devel) gailuron@precision:~/workspace/gudhi/gudhi-devel/build/src/Simplex_tree/benchmark$ make make_filtration_non_decreasing && ./make_filtration_non_decreasing
Building CXX object src/Simplex_tree/benchmark/CMakeFiles/make_filtration_non_decreasing.dir/make_filtration_non_decreasing.cpp.o
Linking CXX executable make_filtration_non_decreasing
Built target make_filtration_non_decreasing
num_simplices= 38001212
... make_filtration_non_decreasing: 10.347s
... make_filtration_non_decreasing: 10.389s
... make_filtration_non_decreasing: 10.588s
... make_filtration_non_decreasing: 10.648s
... make_filtration_non_decreasing: 10.643s
```

# generalization of filtration branch
```txt
(gudhi-devel) gailuron@precision:~/workspace/gudhi/gudhi-devel/build/src/Simplex_tree/benchmark$ make make_filtration_non_decreasing && ./make_filtration_non_decreasing
Building CXX object src/Simplex_tree/benchmark/CMakeFiles/make_filtration_non_decreasing.dir/make_filtration_non_decreasing.cpp.o
Linking CXX executable make_filtration_non_decreasing
Built target make_filtration_non_decreasing
num_simplices= 3799085
... make_filtration_non_decreasing: 0.642s
... make_filtration_non_decreasing: 0.644s
... make_filtration_non_decreasing: 0.645s
... make_filtration_non_decreasing: 0.644s
... make_filtration_non_decreasing: 0.628s
(gudhi-devel) gailuron@precision:~/workspace/gudhi/gudhi-devel/build/src/Simplex_tree/benchmark$ make make_filtration_non_decreasing && ./make_filtration_non_decreasing
Building CXX object src/Simplex_tree/benchmark/CMakeFiles/make_filtration_non_decreasing.dir/make_filtration_non_decreasing.cpp.o
Linking CXX executable make_filtration_non_decreasing
Built target make_filtration_non_decreasing
num_simplices= 38001212
... make_filtration_non_decreasing: 9.763s
... make_filtration_non_decreasing: 9.833s
... make_filtration_non_decreasing: 9.747s
... make_filtration_non_decreasing: 9.76s
... make_filtration_non_decreasing: 9.734s
```

Reply:
Nice unexpected gain!
I think most of the difference comes from initializing the max computation from the old value (instead of computing the max of the faces before comparing it to the old value), which makes comparisons more predictable.
I played a bit with it, and using `!(a<=b)` or `isnan(b)||b<a` didn't make a difference, so we can ignore performance for this discussion.